ironfish/src/wallet/walletdb/transactionValue.ts

```diff
@@ -41,6 +44,14 @@ export class TransactionValueEncoding implements IDatabaseEncoding<TransactionVa
       bw.writeU32(sequence)
     }
 
+    const assetCount = value.assetBalanceDeltas.size
+    bw.writeU32(assetCount)
+
+    for (const [assetIdentifier, balanceDelta] of value.assetBalanceDeltas.entries()) {
+      bw.writeHash(assetIdentifier)
```

Comment:
I don't think this will work, you need to use writeBytes with ASSET_IDENTIFIER. Hash assumes 32 bytes but ASSET_IDENTIFIER is 64 bytes.